src/mavlink/mavlinkmanager.cpp

```diff
@@ -0,0 +1,80 @@
+#include "mavlinkmanager.h"
+
+#include "logger.h"
+
+#include <mavlink_msg_attitude.h>
+
+#include <QDebug>
+#include <QQmlEngine>
+
+PING_LOGGING_CATEGORY(MAVLINKMANAGER, "ping.mavlinkmanager")
+
+MavlinkManager::MavlinkManager()
+{
+    QQmlEngine::setObjectOwnership(this, QQmlEngine::CppOwnership);
+
+    // This is our default udpin connection link
+    // TODO: Allow the user to configure this
+    connect({LinkType::Udp, {"192.168.2.2", "14660"}});
+
+    QObject::connect(&_heartbeatTimer, &QTimer::timeout, this, &MavlinkManager::sendHeartbeatMessage);
+    _heartbeatTimer.start(4000);
+}
+
+void MavlinkManager::connect(const LinkConfiguration& conf)
+{
+    if (_linkIn) {
+        auto link = _linkIn->self();
```

Comment:
the method `self` is quite strange, it sounds like `this` as it's the same thing that python uses.

Reply:
Indeed, we need to rename it. Link is a helper object for the AbstractLink pointer, we can rework that in the future.